rest/config.go

```diff
@@ -284,6 +288,68 @@ type Http2Config struct {
 	Enabled *bool `json:"enabled,omitempty"` // Whether HTTP2 support is enabled
 }
 
+//TODO: Add support for TLS 1.3 when we run Go 1.13
+type TLSVersion uint32
+
+const (
+	TLSv1 TLSVersion = iota
+	TLSv1dot1
+	TLSv1dot2
+
+	tlsvCount
+)
+
+var tlsVersionNames = []string{"tlsv1", "tlsv1.1", "tlsv1.2"}
+
+func (l *TLSVersion) Set(newVersion TLSVersion) {
+	atomic.StoreUint32((*uint32)(l), uint32(newVersion))
```

Comment:
Config handling doesn't typically need to be concurrency-safe - there shouldn't be multiple goroutines attempting to process a config, or to access a TLSVersion instance.